File: C++/evaluate-reverse-polish-notation.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        vector<long> stack;
        
        for (string &t: tokens) {
            if ((t == "+") || (t == "-") || (t == "*") || (t == "/")) {
                long a = stack.back();
                stack.pop_back();
                long b = stack.back();
                stack.pop_back();
                long c = 0;
                if (t == "+") {
                    c = a + b;
                }
                if (t == "-") {
                    c = b - a;
                }
                if (t == "*") {
                    c = a * b;
                } 
                if (t == "/") {
                    c = b / a;
                }
                stack.push_back(c);
            } else {
                stack.push_back(stoi(t));
            }
        }
        return (int)stack.back();
    }
};

File: Kotlin/evaluate-reverse-polish-notation.kt
class Solution {
    var stack: MutableList<Int> = mutableListOf()
    
    fun evalRPN(tokens: Array<String>): Int {
        
        for (token in tokens) {
            when (token) {
                "+" -> push(pop() + pop())
                "*" -> push(pop() * pop())
                "-" -> { var x = pop(); push(pop() - x) }
                "/" -> { var x = pop(); push(pop() / x) }
                else -> push(token.toInt())
            }
        }

        return pop()
    }

    fun push(n: Int) {
        stack.add(n)
    }

    fun pop(): Int {
        var n = stack.last()
        stack.removeAt(stack.size - 1)
        return n
    }
}